### ISSUE TITLE
No DB setup instructions / migrations

### ISSUE THREAD

**Issue by jtxz** [DhivinX/monorepo-starter#2]
This may already be known by people familiar with NestJS / TypeORM but I'm struggling to manually create a DB + table to get the app to run as simply as running `npm run server:dev`. Should there not be a migration of some kind? 🤔 

This is cool project btw, thanks for sharing!

**Comment by DhivinX** [comment]
Hi,

can you show what exactly the problem is? Maybe some screenshots or snippets.

Thank you!

**Comment by jtxz** [comment]
I mean if you just create an empty `postgres` database, point to that in `config.yaml` and then run the app and navigate to `localhost:8080` you get this from the server:
```
{"statusCode":401,"message":"Invalid login data","error":"Unauthorized"}
```

Which is not surprising because we are trying to login with:
```
await userStore.login('admin@admin.com', 'test123');
```

But that user hasn't been created yet, nor has any table to store the users in. I guess there is a step missing in the instructions and probably some migrations needed to create the table and the user?

**Comment by jtxz** [comment]
Ok after doing some reading and debugging I managed to fixed the problem. Due to my inexperience I didn't realise that `synchrronize: true` the table was meant to be created for me. The only issue was that the entity `user` was trying to create a table called `user` which was **failing** because `Postgres` has a built-in table called `user`. This wasn't failing with an error so I had to work this out myself.

Solution in `user.entity.ts`:
```
@Entity("monorepo_user")
```

**I also needed to delete my database and rerun the server.** This meant that a new table was created called `monorepo_user` instead of trying to create one called `user`.

Part 2:
The user `admin@admin.com` still didn't exist so I tried to do this:

```
const user: User = new User();

        user.email = "admin@admin.com";
        user.hash = hashPassword(
            "123456",
            this.configService.get<string>("keys.pwdsalt")
        );
        user.firstName = "Admin";
        user.lastName = "User";

       console.log(user);

        user.save();
```

This fails with:
```
[Nest] 6350  - 06/25/2023, 1:07:09 PM   ERROR [ExceptionHandler] DataSource is not set for this entity.
```

So for now I took the `hash` from the `console.log(user)` and manually inserted the row into the database. I will try to debug the `createUser()` function soon but for now it's authenticating correctly!
